lib/Delighted/Client.php

```
@@ -126,12 +131,13 @@ protected function request($method, $path, $headers = [], $argsOrBody = [])
             $request = new Request($method, $path, $headers);
             $response = $this->adapter->send($request, $argsOrBody);
 
-            return json_decode((string) $response->getBody(), true);
+            return ['json' => json_decode((string) $response->getBody(), true),
+                'headers' => $response->getHeaders()];
         } catch (Exception $e) {
             $r = $e->getResponse();
             $code = $r->getStatusCode();
             $body = [];
-            if (preg_match('#application/json(;|$)#', $r->getHeader('Content-Type')[0])) {
+            if (preg_match('#application/json(;|$)#', $r->getHeader('Content-Type')[0] ?? '')) {
```

Can you elaborate on this change? Apologies in advance if this is a silly question :)

---

Not silly, intent was to suppress this exception during testing, as our phpunit is configured to convert notices (in this case a possible indefined index `[0]`) to exceptions. Forgot I left this there, not needed, cleaner way is to properly add the `Content-Type` header in our tests and keep notices output.